Document notification center (#456)
discord-userdoccers/discord-userdoccers: pages/resources/notification-center.mdx

| id                       | snowflake                                                                | The ID of the notification center item                                      |
| bundle_id                | string                                                                   | Unknown                                                                     |
| type                     | string                                                                   | The [type of notification center item](#notification-center-item-type)      |
| body                     | string                                                                   | The body text of the notification center item                               |
| acked                    | boolean                                                                  | Whether the notification center item is acknowledged                        |
| deeplink                 | string                                                                   | The URL used for deep linking                                               |

> | icon_url                 | ?string                                                                  | The icon URL of the item                                                    |

Reviewer: ```suggestion
| icon_url                 | ?string                                                                  | The URL of the icon to display                                                   |
```

| message                  | ?[message](/resources/message#message-object) object                     | The message associated with the item                                        |
| item_enum                | ?integer                                                                 | The [item enum of notification center item](#notification-center-item-enum) |
| completed                | boolean                                                                  | Whether the item is completed                                               |
| guild_id                 | ?snowflake                                                               | The ID of the guild associated with the item                                |
| icon_name                | ?string                                                                  | Unknown                                                                     |
| message_id               | ?snowflake                                                               | The ID of the message associated with the item                              |

> | message_channel_id       | ?snowflake                                                               | The ID of the channel of message associated with the item                   |

Reviewer: ```suggestion
| message_channel_id       | ?snowflake                                                               | The ID of the channel of the message associated with the item                   |
```

###### Notification Center Item Type

| Value                         | Description                                                                 |
| ----------------------------- | --------------------------------------------------------------------------- |
| go_live_push                  | Unknown                                                                     |
| friend_request_accepted       | Outgoing friend request was accepted                                        |
| friend_request_pending        | A friend request is pending                                                 |
| friend_suggestion_created     | A friend suggestion is created                                              |
| friend_request_reminder       | Reminder about pending friend requests                                      |
| dm_friend_nudge               | Reminder to message friends                                                 |
| recent_mention                | Recent mention                                                              |
| reply_mention                 | Mention through reply                                                       |

> | scheduled_guild_event_started | A scheduled event in guild is started                                       |

Reviewer: ```suggestion
| scheduled_guild_event_started | A guild scheduled event started                                       |
```

> | system_demo                   | Unknown                                                                     |

Reviewer: ```suggestion
| system_demo                   | Demo notification |
```

| missed_messages               | The current user missed messages in frequently read private channels/guilds |
| top_messages                  | Unknown                                                                     |

> | lifecycle_item                | Unknown                                                                     |

Reviewer: ```suggestion
| lifecycle_item                | New user tutorial item |
```

| ----- | -------------- | -------------- |
| 0     | UPDATE_PROFILE | Update profile |
| 1     | FIND_FRIENDS   | Find friends   |
| 2     | ADD_FRIEND     | Add friend     |
| 3     | FIRST_MESSAGE  | First message  |

<RouteHeader method="GET" url="/users/@me/notification-center/items">
  Get Notification Center Items
</RouteHeader>

Returns items that the current user has in notification center.

###### Query String Params

| Field            | Type      | Description                                                                                                     |
| ---------------- | --------- | --------------------------------------------------------------------------------------------------------------- |
| after?           | snowflake | Get notification center items after this notification center item ID                                            |
| with_mentions?   | boolean   | Whether to include notification center items that directly mention the current user                             |
| roles_filter?    | boolean   | Whether to include notification center items that mention the current user through role                         |

> | everyone_filter? | boolean   | Whether to include notification center items that mention the current user through @everyone and @here mentions |

Reviewer: ```suggestion
| with_mentions?   | boolean   | Whether to include recent mention notifications (default false) |
| roles_filter?    | boolean   | Whether to include role mentions (default true)                         |
| everyone_filter? | boolean   | Whether to include @everyone and @here mentions (default true) |
```

| has_more | boolean                                                                    | Whether there are potentially additional notification center items that could be returned on a subsequent call |

<RouteHeader method="DELETE" url="/users/@me/notification-center/items/{notification_center_item.id}">
  Delete Notification Center Item
</RouteHeader>

Deletes a notification center item. Returns a 204 empty response on success. Fires [Notification Center Item Delete](/topics/gateway-events#notification-center-item-delete) Gateway event.

###### JSON Params

| Field      | Type   | Description                                                                              |
| ---------- | ------ | ---------------------------------------------------------------------------------------- |
| item_type? | string | The [deletion type of notification center item](#notification-center-item-deletion-type) |

###### Notification Center Item Deletion Type

| Value   | Description                |
| ------- | -------------------------- |

> | mention | The item is a mention item |

Reviewer: ```suggestion
| mention | The item is a mention (`type` of [`reply_mention` or `recent_mention`](#notification-center-item-type)) |
```
